example/03-gossip/utility.cpp

```diff
@@ -52,7 +52,8 @@ namespace libp2p::protocol::example::utility {
     boost::asio::ip::tcp::resolver resolver(io);
     boost::asio::ip::tcp::resolver::query query(boost::asio::ip::host_name(),
                                                 "");
-    boost::asio::ip::tcp::resolver::iterator it = resolver.resolve(query);
+    boost::system::error_code ec;
+    boost::asio::ip::tcp::resolver::iterator it = resolver.resolve(query, ec);
```

Comment:
`ec` is unused. Perhaps it might good to probe it if error have happened.

Reply:
`ec` prevents `resolve` exception.
`getLocalIP` just returns `127.0.0.1` if host don't have name or can't `resolve`.
If more complicated code to enumerate network interfaces (like https://stackoverflow.com/a/62303963) required, it should be added elsewhere and reused.